sentinelhub/download/client.py

```diff
@@ -135,8 +156,11 @@ def _single_download(self, request, decode_data):
 
     @retry_temporary_errors
     @fail_user_errors
-    def _execute_download(self, request):
+    def _execute_download(self, request: DownloadRequest) -> bytes:
         """A default way of executing a single download request"""
+        if request.url is None:
+            raise RuntimeError(f"Faulty request {request}, no URL specified.")
```

Comment:
I would actually suggest raising `ValueError` here because this error could only happen if a user provides wrong input parameters. `RuntimeError` is more natural for cases where the reason for an error is e.g. that SH service returned an unexpected output such as in such case:

```Python
        if not isinstance(json_response, dict) or "data" not in json_response:
            raise RuntimeError(f"Response from {url} did not contain the expected information.")
```

Reply:
Was on the verge regarding the kind of exceptions. Since we often configure things on the fly (though we shouldn't) I thought runtime wouldn't be worst choice. Will adjust the rest.